pkg/cache/v2/snapshot.go

```diff
@@ -194,3 +209,44 @@ func (s *Snapshot) GetVersion(typeURL string) string {
 	}
 	return s.Resources[typ].Version
 }
+
+// GetVersionMap will construct a verison map of the current snapshot
+func (s *Snapshot) GetVersionMap() (map[string]map[string]string, error) {
+	if s == nil {
+		return nil, fmt.Errorf("missing snapshot")
+	}
+
+	// We need to initialize the map before we process anything since it is nested
+	versionMap := initializeVMap()
+	for i := 0; i < int(types.UnknownType); i++ {
+		typeURL := GetResponseTypeURL(types.ResponseType(i))
+		resources := s.GetResources(typeURL)
```

Comment:
Can we just iterate over the keys in the snapshot items instead? Seems like this is a bit of a roundabout way of getting to the resource names

Reply:
Yup, just changed, good idea thank you for the catch